### ISSUE TITLE
Event path and status code are wrong in Nitro hooks

### ISSUE THREAD

**Issue by antoinerey** [nuxt/nuxt#26113]
### Environment

(Edit: I've updated the reproduction to use the `error` hook instead of `afterResponse`)

------------------------------
- Operating System: Darwin
- Node Version:     v20.11.0
- Nuxt Version:     3.10.3
- CLI Version:      3.10.1
- Nitro Version:    2.8.1
- Package Manager:  yarn@4.1.0
- Builder:          -
- User Config:      app, extends, runtimeConfig, modules, sourcemap, vue, build
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

https://stackblitz.com/edit/nuxt-starter-pa68ta

Here is a minimal Nuxt application with only two modifications:

1. I make sure to throw an error from the `app.vue` file.
2. I register a new Nitro hook to log both the event path and status code (via a Nitro plugin).

### Describe the bug

Given the setup above, when I load `/`, I expect two things to be logged:

1. `/` as the event path.
2. `500` as the event status code.

However, this is not what I receive. Instead, I get:

1. `/__nuxt_error?...` as event path.
2. `200` as event status.

### Additional context

The overall idea is to trace incoming requests, generate APM spans, and send them to Datadog to monitor the server's health. Each span is tagged with a couple of properties, including both the event path and status code.

Doing it inside a Nitro hook allows me to monitor both HTML and API requests from a single place.

It roughly looks like this in real life:

```ts
app.hooks.hook('request', (event) => {
  const { req } = event.nod

  const span = tracer.startSpan('web.request')

  span.setTag('resource.name', `${req.method} ${req.url}`)
  span.setTag('http.method', req.method)
  span.setTag('http.url', req.url)

  event.context.span = span
})

app.hooks.hook('error', (error, { event }) => {
  const { res } = event.node

  if (res.statusCode >= 500) {
    event.context.span.setTag('error', true)
    event.context.span.setTag('sampling.priority', 1)
  }

  event.context.span.setTag('http.status_code', res.statusCode)
  event.context.span.finish()
})
```

**Comment by antoinerey** [comment]
So... I dug a bit deeper into h3, trying to reproduce the issue in an even more minimal way, but it looks like `onError` and `onAfterResponse` hooks are not fired when an error is thrown from a handler.

```ts
export const app = createApp({
  onRequest(event) {
    console.log('onRequest');
    console.log(event.path);
    console.log(getResponseStatus(event));
  },

  onError(error, event) {
    console.log('onError');
    console.log(event.path);
    console.log(getResponseStatus(event));
  },

  onBeforeResponse(event) {
    console.log('onBeforeResponse');
    console.log(event.path);
    console.log(getResponseStatus(event));
  },

  onAfterResponse(event) {
    console.log('onAfterResponse');
    console.log(event.path);
    console.log(getResponseStatus(event));
  },
});

app.use(
  '/',
  eventHandler((event) => {
    throw createError({
      message: 'Oops',
      statusCode: 500,
    });
  })
);
```

Loading `/` does execute the `onRequest` hook as shown in the logs, but nothing more. It looks like we can't get notified of errors via h3 hooks, which might explain why I don't get proper results in a Nuxt application either. I'm not deeply familiar with h3, so am I missing something here?

The complete reproduction is here: https://stackblitz.com/edit/unjs-h3-bql2cu.

**Comment by danielroe** [comment]
Would you see if you can reproduce this in pure Nitro ([reproduction sandbox](https://stackblitz.com/github/unjs/nitro/tree/main/examples/hello-world)), and if so, raise [there](https://github.com/unjs/nitro/issues/new/choose)? 🙏

**Comment by antoinerey** [comment]
Here you go: https://github.com/unjs/h3/issues/695. 🤗

**Comment by antoinerey** [comment]
Following the other issue on h3, I figured that I was not using the proper hook. Instead of using the `afterResponse` hook to catch errors, I should be using the `error` hook. My understanding is that the response hooks (`beforeResponse` and `afterResponse`) are not called when an error happens. 

So, I've updated the reproduction, and one of the issues is gone: the path is now correct. But, the event status code is still 200 whereas it should be 500. There is a workaround to fetch the proper status code from the error object, but I believe there's something wrong with the event status code.

Below is the updated reproduction. (Note that the initial reproduction link still works)

```ts
export default defineNitroPlugin((nitro) => {
  nitro.hooks.hook('error', (error, { event }) => {
    console.log('onError');
    console.log(event.path);                                 // `/` 
    console.log('event status', getResponseStatus(event));   // `200` <--
    console.log('error status', error.statusCode);           // `500` <--
  });
});
```
